frontend/views/containers/GroupPledgesGraph.vue

```diff
@@ -1,218 +1,204 @@
       tooltip(
-        v-if='graphData.userIncomeNeeded !== graphData.userIncomeToReceive'
-        direction='right'
+        v-if='graphData.ourIncomeNeeded !== graphData.ourIncomeToReceive'
+        :hasTextCenter='true'
+        :text='L("Based on other members pledges, the group is not able to provide a full mincome yet.")'
       )
-        | {{fakeStore.currency}}{{graphData.userIncomeToReceive}}
-        i.icon-info-circle.is-size-6.has-text-tertiary.c-legendItem-icon
-
-        template(slot='tooltip')
-          i18n.has-text-weight-bold(tag='strong') Income Incomplete
-          i18n.has-text-weight-normal(
-            tag='p'
-            :args='{ amount: `${fakeStore.currency}${graphData.userIncomeNeeded}` }'
-          )
-            | The group at the moment is not pledging enough to cover everyone&apos;s mincome.
-            | So you&apos;ll receive only a part instead of the {amount} you need.
-
-      template(v-else='')
-        | {{fakeStore.currency}}{{graphData.userIncomeToReceive}}
+        i.icon-info-circle.is-suffix.has-text-primary
 </template>
 
 <script>
-import { PieChart, GraphLegendGroup, GraphLegendItem } from '@components/Graphs/index.js'
+import { mapGetters } from 'vuex'
+import { PieChart, GraphLegendItem } from '@components/Graphs/index.js'
 import Tooltip from '@components/Tooltip.vue'
 import currencies from '@view-utils/currencies.js'
-import { debounce } from '@utils/giLodash.js'
-import { TABLET } from '@view-utils/breakpoints.js'
+import distributeIncome from '@utils/distribution/mincome-proportional.js'
 
 export default {
   name: 'GroupPledgesGraph',
   components: {
     PieChart,
-    GraphLegendGroup,
     GraphLegendItem,
     Tooltip
   },
   props: {
-    // Needed to have a dynamic graph (ex: used on Contributions page)
-    userPledgeAmount: {
-      type: Number,
+    type: {
+      type: String, // incomeAmount || pledgeAmount TODO validator
       default: null
     },
-    userIncomeAmount: {
+    amount: {
       type: Number,
       default: null
     }
   },
-  data () {
-    return {
-      // REVIEW - make this a watcher
-      isMobile: this.verifyIsMobile(),
-      // -- Hardcoded Data just for layout purpose:
-      fakeStore: {
-        currency: currencies.USD.symbol,
-        groupMembersTotal: 7,
-        groupPledgeGoal: 800,
-        mincome: 500,
-        othersPledges: [180, 130, 200]
-      }
-    }
-  },
-  mounted: function () {
-    window.addEventListener('resize', this.handleResize)
-  },
-  beforeDestroy: function () {
-    window.removeEventListener('resize', this.handleResize)
-  },
   computed: {
-    chartSize () {
-      return this.isMobile ? '8rem' : undefined
-    },
-    legendItemClass () {
-      return 'column c-legend-item'
+    ...mapGetters([
+      'groupSettings',
+      'groupProfiles',
+      'groupMembersCount',
+      'ourIdentityContractId',
+      'ourUsername'
+    ]),
+    doWePledge () {
+      return this.type === 'pledgeAmount'
```

Comment:
Note: since this function is only called from within `graphData()`, it is better to keep this as a variable in that function. A function call can add significant overhead, and the `graphData()` function has significant computation in it. So anything unnecessary should be removed. So I recommend defining this as a `doWeNeedPledge` const inside `graphData()`